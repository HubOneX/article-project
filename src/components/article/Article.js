import Body from "./body/Body";
import Header from "./header/Header";
import MainImage from "./mainImage/MainImage";
import NotLoaded from "./notLoaded/NotLoaded";
import ArticleWrapper from "./wrapper/ArticleWrapper";

const Article = ({ article, isError }) => {
  if (isError || !article) {
    return <NotLoaded />;
  }
  const { author, heading, date, image, body } = article;

  return (
    <ArticleWrapper>
      <MainImage image={image} />
      <Header heading={heading} publicationDate={date} author={author} />
      <Body bodyElements={body} />
    </ArticleWrapper>
  );
};

export default Article;
