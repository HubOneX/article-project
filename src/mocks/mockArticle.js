import { urls } from "../constants/urls";

export const mockArticle = {
  author: "Stephen King",
  heading: "The horror of tests",
  date: "2020-09-03T07:27:59.490",
  image: { url: urls.globalUrl + "/url", caption: "main image" },
  body: ["<h2>Secondary heading</h2>", "<p>The paragraph below!</p>"],
};

export const mockArticleEmptyValues = {
  author: "",
  heading: "",
  date: "",
  image: { url: "", caption: "" },
  body: "",
};

export const mockArticleResponse = {
  data: {
    elements: {
      author: { value: "Stephen King" },
      heading: { value: "The horror of tests" },
      date: { value: "2020-09-03T07:27:59.490" },
      mainImage: {
        value: {
          leadImage: { renditions: { lead: { source: "/url" } } },
          leadImageCaption: { value: "main image" },
        },
      },
      body: {
        values: ["<h2>Secondary heading</h2>", "<p>The paragraph below!</p>"],
      },
    },
  },
};
