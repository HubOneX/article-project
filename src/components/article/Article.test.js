import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Article from "./Article";
import { mockArticle } from "../../mocks/mockArticle";

describe("Article", () => {
  it("should return NotLoaded element if an error is provided", () => {
    render(<Article isError article={mockArticle} />);
    const notLoaded = screen.getByText(
      "Sorry, we couldn't load this article. Please try again later."
    );

    expect(notLoaded).toBeInTheDocument();
  });

  it("should return NotLoaded element if the artcile is not provided", () => {
    render(<Article isError={false} article={undefined} />);
    const notLoaded = screen.getByText(
      "Sorry, we couldn't load this article. Please try again later."
    );

    expect(notLoaded).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const article = renderer.create(
      <Article article={mockArticle} isError={false} />
    );

    expect(article).toMatchSnapshot();
  });
});
