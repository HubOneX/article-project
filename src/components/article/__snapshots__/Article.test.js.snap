// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should match the snapshot 1`] = `
<div
  className="article-wrapper"
>
  <img
    alt="main image"
    className="article-image"
    src="https://content-eu-4.content-cms.com/api/859f2008-a40a-4b92-afd0-24bb44d10124/url"
  />
  <div
    className="article-header"
  >
    <div
      className="article-info"
    >
      <p
        className="article-author"
      >
        Stephen King
      </p>
      <p
        className="article-date"
      >
        07:27 Sep 3, 2020
      </p>
    </div>
    <hr />
    <h1
      className="article-title"
    >
      The horror of tests
    </h1>
  </div>
  <div
    className="article-body"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>Secondary heading</h2>",
        }
      }
    />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>The paragraph below!</p>",
        }
      }
    />
  </div>
</div>
`;
